lib/webbluetooth/bindings.js

```diff
@@ -0,0 +1,329 @@
+var events = require('events');
+var util = require('util');
+
+var debug = require('debug')('bindings');
+
+var ble = navigator.bluetooth;
+
+
+function addDashes(uuid){
+  if(!uuid){
+    return uuid;
+  }
+  if(uuid && uuid.length === 32){
+    uuid = uuid.substring(0,8) + '-' + uuid.substring(8,12) + '-' +  uuid.substring(12,16) + '-' +  uuid.substring(16,20) + '-' +  uuid.substring(20);
+  }
+  return uuid.toLowerCase();
+}
+
+function stripDashes(uuid){
+  if(uuid){
+    uuid = uuid.split('-').join('');
+  }
+  return uuid;
+}
+
+
+var NobleBindings = function() {
+
+  this._startScanCommand = null;
+  this._peripherals = {};
+
+  var self = this;
+
+  process.nextTick(function(){
+    debug('first state change');
+    self.emit('stateChange', 'poweredOn');
+  });
+
+};
+
+util.inherits(NobleBindings, events.EventEmitter);
+
+NobleBindings.prototype.init = function() {
+  var self = this;
+  // no-op
+  setTimeout(function(){
+    debug('initing');
+    self.emit('stateChange', 'poweredOn');
+  }, 2000);
+};
+
+
+NobleBindings.prototype._onOpen = function() {
+  debug('on -> open');
+};
+
+NobleBindings.prototype._onClose = function() {
+  debug('on -> close');
+
+  this.emit('stateChange', 'poweredOff');
+};
+
+NobleBindings.prototype.startScanning = function(serviceUuids, allowDuplicates) {
+  var self = this;
+  debug('startScanning', serviceUuids, allowDuplicates);
+
+  if(!Array.isArray(serviceUuids)){
+    serviceUuids = [serviceUuids];
+  }
+
+  ble.requestDevice({ filters: [{ services: serviceUuids.map(addDashes) }] })
+    .then(function(device){
+      debug('scan finished', device);
+      if(device){
+
+        var address = device.id;
+        //TODO use device.adData when api is ready
+        //rssi = device.adData.rssi;
+
+        self._peripherals[address] = {
+          uuid: address,
+          address: address,
+          advertisement: {localName:device.name}, //advertisement,
+          rssi: device.adData.rssi,
+          device: device,
+          cachedServices: {},
+          localName: device.name,
+          serviceUuids: serviceUuids
+        };
+
+        // self.emit('discover', device.id, self._peripherals[address], device.adData.rssi);
+        self.emit('discover', device.id, device.id, device.addressType, !device.paired, self._peripherals[address].advertisement, self._peripherals[address].rssi);
+
+      }
+    }, function(err){
+      debug('err scanning', err);
+    });
+
+  this.emit('scanStart');
+};
+
+NobleBindings.prototype.stopScanning = function() {
+  this._startScanCommand = null;
+
+  //TODO: need web api completed for this to work'=
+  this.emit('scanStop');
+};
+
+NobleBindings.prototype.connect = function(deviceUuid) {
+  var self = this;
+  debug('connect', deviceUuid);
+  var peripheral = this._peripherals[deviceUuid];
+
+  // Attempts to connect to remote GATT Server.
+  peripheral.device.gatt.connect()
+    .then(function(gattServer){
+      peripheral.gattServer = gattServer;
```

Comment:
Do you really need `gattServer` now since it is always defined in `peripheral.device` as `peripheral.device.gatt`?